### PR TITLE
Side 2011 add cursor aptitude

### PR DESCRIPTION
**Pull request by RusGraf**

### CODE REVIEW

File: ldk/goja/src/clipboard.ts [hunk 1]
```diff
@@ -1,28 +1,47 @@
...
 }
 
 function read(): Promise<string> {
   return new Promise<string>((resolve, reject) => {
     try {
-      oliveHelps.clipboard.read((val: string) => resolve(val));
+      oliveHelps.clipboard.read((clipboardText: string) => resolve(clipboardText));
     } catch (e) {
       reject(e);
       // TODO: add console log
```
Comment (cshields-oliveai) [inline ldk/goja/src/clipboard.ts:36]: did we want to develop some very basic standard for a console log for these here?

Comment (RusGraf) [reply ldk/goja/src/clipboard.ts]: I agree, pushed changes.

**Review by cshields-oliveai — APPROVED** [review]

File: ldk/goja/.eslintrc.js [hunk 2]
```diff
@@ -20,6 +20,7 @@ module.exports = {
     'class-methods-use-this': 'off',
     'import/no-extraneous-dependencies': ['error', { devDependencies: true }],
     'import/prefer-default-export': 'off', // Default exports are bad.
+    'no-console': 'off',
```
Comment (cshields-oliveai) [inline ldk/goja/.eslintrc.js:23]: It has occurred to me that these console logs will appear in our unit test output as the rejection tests execute. We will need to come up with a more reliable way to do these logs.

File: ldk/goja/src/clipboard.ts [hunk 3]
```diff
@@ -1,30 +1,50 @@
...
+   * @returns A Promise resolving with the current contents of the clipboard.
+   */
   read(): Promise<string>;
-  write(val: string): Promise<void>;
-  listen(cb: (val: string) => void): void;
+
+    /**
+   * Writes the provided text into the clipboard.
+   *
+   * @param text
```
Comment (cshields-oliveai) [inline ldk/goja/src/clipboard.ts:14]: description of this param?

**Review by cshields-oliveai — APPROVED** [review]

File: ldk/goja/.eslintrc.js (hunk 2 above)
Comment (richseviora) [reply ldk/goja/.eslintrc.js:23]: I wonder if we can mock the console functions... if so we can do that before each test (in a shared before each or something)

**Review by richseviora — APPROVED** [review]

File: ldk/goja/.eslintrc.js (hunk 2 above)
Comment (RusGraf) [reply ldk/goja/.eslintrc.js:23]: We could abstract console perhaps...

Comment (cshields-oliveai) [reply ldk/goja/.eslintrc.js:23]: Could potentially mock console.log() globally like with olivehelps. I could give this a shot with the window aptitude card - if it doesn't pan out I will make a task for us to figure out a nicer way.

**Pull request merged**